Documentation/Enums/Enum Declarations.md

```diff
@@ -0,0 +1,223 @@
+
+`enum start` `:` `name`
+
+Starts declaring an enum type.
+
+### `enum member`
+
+`enum member` `:` `name value`*`[, ","]`*
+
+Declares a value of an enum inside its declaration. The value must be a numeric.
+
+A comma may end the line to indicate this is not the last enum member in its declaration. 
+
+
+## Usage
+
+```
+enum start : Direction
+    enum member : Unknown 0 ,
+    enum member : Horizontal 1 ,
+    enum member : Vertical 2
+enum end
+```
+
+### CSharp
+
+```
+enum Direction {
+    Unknown = 0,
+    Horizontal = 1,
+    Vertical = 2
+}
+```
+
+### Java
+
+```
+enum Direction {
+    Unknown (0),
+    Horizontal (1),
+    Vertical (2)
+}
+```
+
+### Python
+
+```
+from enum import Enum
+
+class Direction(Enum):
+    Unknown = 0
+    Horizontal = 1
+    Vertical = 2
+```
+
+### Ruby
+
+```
+class Direction
+    Unknown = 0
+    Horizontal = 1
+    Vertical = 2
+end
+```
+
+### TypeScript
+
+```
+enum Direction {
+    Unknown = 0,
+    Horizontal = 1,
+    Vertical = 2
+}
+```
+
+## Implementation
+
+An enum declaration consists of a starting line, any number of values, and an endling line.
+
+The starting line consists of `DeclareStartLeft`, the provided name, then `DeclareStartRight`.
+
+Each value line consists of a member name, `DeclareValueLeft`, the value, and `DeclareValueRight`.
+Value lines with a comma appended end in `DeclareCommaRight`; value lines without end in `DeclareLastRight`.
+
+The ending line consists solely of `DeclareEnd`.
+
+### Properties
+
+<table>
+    <thead>
+        <th>Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </thead>
+    <tbody>
+        <tr>
+            <td>DeclareStartLeft</td>
+            <td>string</td>
+            <td>Start of the first line of an enum declaration.</td>
+        </tr>
+        <tr>
+            <td>DeclareStartRight</td>
+            <td>string</td>
+            <td>End of the first line of an enum declaration.</td>
+        </tr>
+        <tr>
+            <td>DeclareValueLeft</td>
+            <td>string</td>
+            <td>Start of a line declaring an enum member and value.</td>
+        </tr>
+        <tr>
+            <td>DeclareValueMiddle</td>
+            <td>string</td>
+            <td>Middle of a line declaring an enum member and value.</td>
+        </tr>
+        <tr>
+            <td>DeclareValueRight</td>
+            <td>string</td>
+            <td>End of a line declaring an enum member and value.</td>
+        </tr>
+        <tr>
+            <td>DeclareCommaRight</td>
+            <td>string</td>
+            <td>Suffix after a member and value declaration.</td>
+        </tr>
+        <tr>
+            <td>DeclareLastRight</td>
+            <td>string</td>
+            <td>Suffix after the last member and value declaration in an enum.</td>
+        </tr>
+        <tr>
+            <td>DeclareEnd</td>
+            <td>string</td>
+            <td>End line of an enum declaration.</td>
+        </tr>
+    </tbody>
+</table>
```

Comment:
I get why we're using the html table, but I did like being able to see the table in ascii format as well in the plain text :/